src/data/set/lattice.lean

```diff
@@ -57,6 +62,20 @@ variables {f : α → β}
 protected lemma image_preimage : galois_connection (image f) (preimage f) :=
 assume a b, image_subset_iff
 
+protected lemma push_pull {α : Type*} {β : Type*} (f : α → β) (s : set α) (t : set β) :
+f '' (s ∩ f ⁻¹' t) = f '' s ∩ t :=
```

Comment:
I would never guess this name, and `data/set/basic` looks like a better place for this lemma.

Reply:
This is the math name (see eg. the first google hit [here](https://mathoverflow.net/questions/18799/ubiquity-of-the-push-pull-formula)) but it can go in the docstring for greppability. What name would you like?

Reply:
My bad. Though may be some other mathlib users might have the same problem. I failed to come up with a good `mathlib`-style name. @sgouezel ?